core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+                  </a>
+                </li>
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/notifications_dashboard"
+                     class="protractor-test-notifications-link">
+                    <span translate="I18N_TOPNAV_NOTIFICATIONS"></span>
+                    <span ng-if="numUnseenNotifications > 0">
+                      (<[numUnseenNotifications]>)
+                    </span>
+                  </a>
+                </li>
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/preferences"
+                     class="protractor-test-preferences-link">
+                    <span translate="I18N_TOPNAV_PREFERENCES"></span>
+                  </a>
+                </li>
+
+                {% if is_moderator %}
+                  <li>
+                    <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/moderator" target="_blank">
+                      <span translate="I18N_TOPNAV_MODERATOR_PAGE"></span>
+                    </a>
+                  </li>
+                {% endif %}
+                {% if is_super_admin %}
+                  <li>
+                    <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/admin" target="_blank">
+                      <span translate="I18N_TOPNAV_ADMIN_PAGE"></span>
+                    </a>
+                  </li>
+                {% endif %}
+                <hr class="oppia-top-right-menu-item-separator">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="{{logout_url}}">
+                    <span translate="I18N_TOPNAV_LOGOUT"></span>
+                  </a>
+                </li>
+              </ul>
+            </li>
+          {% else %}
+            <li class="dropdown oppia-navbar-clickable-dropdown pull-right">
+              <div class="oppia-navbar-button-container" style="margin-right: 10px;">
+                <button class="btn oppia-navbar-button"
+                        ng-click="onLoginButtonClicked('{{login_url}}')">
+                  <span translate="I18N_TOPNAV_SIGN_IN"></span>
+                  <span class="caret"></span>
+                </button>
+              </div>
+              <ul class="dropdown-menu oppia-navbar-dropdown" role="menu" style="margin-right: 15px; padding: 0;" ng-mouseover="onMouseoverDropdownMenu($event)" ng-mouseleave="onMouseoutDropdownMenu($event)">
+                <li>
+                  <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked('{{login_url}}')">
+                    <img ng-src="<[getStaticImageUrl('/signin/Red-signin-Long-base-44dp.png')]>">
+                  </a>
+                </li>
+              </ul>
+            </li>
+          {% endif %}
+        </ul>
+        <ul class="nav oppia-navbar-nav" ng-if="['create', 'explore', 'collection'].indexOf(NAV_MODE) === -1">
```

Comment:
This looks like something that could be split off to its own directive. Also, the ``ng-if`` logic really ought to be a function in the directive that is called. Plus, the array is clearly full of special nav mode values and should probably be a constant.